src/components/datepicker/Datepicker.vue

```diff
@@ -680,9 +677,16 @@ export default {
         */
         onChangeNativePicker(event) {
             const date = event.target.value
-            this.computedValue = date ? new Date(date + 'T00:00:00') : null
+            const s = date ? date.split('-') : []
+            if (s.length === 3) {
+                const year = parseInt(s[0], 10)
+                const month = parseInt(s[1]) - 1
+                const day = parseInt(s[2])
+                this.computedValue = new Date(year, month, day, 12)
```

Comment:
Why 12 ? It isn't a correct date. Keep in mind that value will be emitted to parent component

Reply:
12 makes it noon as opposed to midnight. If folks are relying on a specific time of day within the emitted Dates, then I suppose that's problematic. The issue is that Safari is broken in such a way that you can't reliably construct a date with midnight local time when the date in question predates the most recent set of Daylight Savings rules. For instance, in the USA, we last changed DST rules in 2006, so dates constructed within range of what is now DST but was previously Standard Time will be off by an hour:

> new Date(2020,3,1).toLocaleDateString()
> "4/1/2020"
> new Date(2005,3,1).toLocaleDateString()
> "3/31/2005"
> new Date(2005,3,1).toLocaleString()
> "3/31/2005, 11:00:00 PM"

Constructing them at noon instead of midnight works around this. 

Chrome had this issue as well for many years, but it was resolved a year or two ago. 

There's a good reason most date controls emit ISO 8601 strings instead of JS Date objects.

Reply:
It's a breaking change because in this way the date object has a different time part and I don't know if it's really correct to get a not 00:00 time

Reply:
It doesn't, unfortunately. It's only once you do a locale format on it that you can see a problem. 

I've updated the PR to construct the dates against midnight again; given we're emitting a Date object this seems like the least worst option. Most consumers of that Date object will (hopefully) be interrogating it to get the numeric date elements (where it behaves correctly) as opposed to locale formatting it, so this should be the least surprising implementation.